fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryManager.java

```diff
@@ -73,6 +87,70 @@ public synchronized void close() throws IOException {
         createdDirectories.clear();
     }
 
+    @SuppressWarnings("PMD.CloseResource")
+    public CompletableFuture<Void> mergeIndex(LuceneAnalyzerWrapper analyzerWrapper) {
+
+        final ScanProperties scanProperties = ScanProperties.FORWARD_SCAN.with(
+                props -> props.clearState().setReturnedRowLimit(1));
+        final Range range = state.indexSubspace.range();
+        final KeyRange keyRange = new KeyRange(range.begin, range.end);
+
+        final RecordCursor<Tuple> cursor = new ChainedCursor<>(
+                state.context,
+                lastKey -> nextTuple(state.context, state.indexSubspace, keyRange, lastKey, scanProperties),
+                Tuple::pack,
+                Tuple::fromBytes,
+                null,
+                scanProperties);
+
+        return cursor.forEach(groupingKey -> mergeIndex(analyzerWrapper, groupingKey));
+    }
+
+    private void mergeIndex(LuceneAnalyzerWrapper analyzerWrapper, Tuple groupingKey) {
+        try {
+            getDirectoryWrapper(groupingKey).mergeIndex(analyzerWrapper);
+        } catch (IOException e) {
+            if (LOG.isWarnEnabled()) {
+                LOG.warn("Lucene mergeIndex failed: error={}", e.getMessage());
```

Comment:
This is not the proper way to log information in the record layer, it should use `KeyValueLogMessage`.
Also, when logging exceptions, the exception should be added as the second parameter to the logging call, so that the stack trace is included.

Reply:
✅